lib/parachain/validation_protocol.go

```diff
@@ -8,23 +8,199 @@ import (
+
+	// The signature by the validator of the signed payload.
+	Signature ValidatorSignature `scale:"3"`
+}
+
+type Bitfield struct {
+	Hash                                common.Hash                         `scale:"1"`
+	UncheckedSignedAvailabilityBitfield UncheckedSignedAvailabilityBitfield `scale:"2"`
+}
+
+// Index returns the VaryingDataType Index
+func (Bitfield) Index() uint {
+	return 0
+}
+
+type BitfieldDistributionMessage scale.VaryingDataType
+
+func NewBitfieldDistributionMessageVDT() BitfieldDistributionMessage {
+	vdt := scale.MustNewVaryingDataType(Bitfield{})
+	return BitfieldDistributionMessage(vdt)
+}
+
+func (BitfieldDistributionMessage) New() BitfieldDistributionMessage {
+	return NewBitfieldDistributionMessageVDT()
+}
+
+// Set will set a VaryingDataTypeValue using the underlying VaryingDataType
+func (bdm *BitfieldDistributionMessage) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*bdm)
+	err = vdt.Set(val)
+	if err != nil {
+		return fmt.Errorf("setting value to varying data type: %w", err)
+	}
+
+	*bdm = BitfieldDistributionMessage(vdt)
+	return nil
+}
+
+// Value returns the value from the underlying VaryingDataType
+func (bdm *BitfieldDistributionMessage) Value() (scale.VaryingDataTypeValue, error) {
+	vdt := scale.VaryingDataType(*bdm)
+	return vdt.Value()
+}
+
+type BitfieldDistribution BitfieldDistributionMessage
+
+func NewBitfieldDistributionVDT() BitfieldDistribution {
+	vdt := scale.MustNewVaryingDataType(Bitfield{})
+	return BitfieldDistribution(vdt)
+}
+
+func (BitfieldDistribution) Index() uint {
+	return 1
+}
+func (BitfieldDistribution) New() BitfieldDistribution {
+	return NewBitfieldDistributionVDT()
+}
+
+// Set will set a VaryingDataTypeValue using the underlying VaryingDataType
+func (bd *BitfieldDistribution) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*bd)
+	err = vdt.Set(val)
+	if err != nil {
+		return fmt.Errorf("setting value to varying data type: %w", err)
+	}
+
+	*bd = BitfieldDistribution(vdt)
+	return nil
+}
+
+type ApprovalDistribution ApprovalDistributionMessage
+
+// NewApprovalDistributionMessageVDT ruturns a new ApprovalDistributionMessage VaryingDataType
+func NewApprovalDistributionVDT() ApprovalDistribution {
+	vdt, err := scale.NewVaryingDataType(Assignments{}, Approvals{})
+	if err != nil {
+		panic(err)
+	}
+	return ApprovalDistribution(vdt)
+}
+
+// New returns new ApprovalDistributionMessage VDT
+func (ApprovalDistribution) New() ApprovalDistribution {
+	return NewApprovalDistributionVDT()
+}
+
+func (ApprovalDistribution) Index() uint {
+	return 4
+}
+
+// Set will set a VaryingDataTypeValue using the underlying VaryingDataType
+func (ad *ApprovalDistribution) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*ad)
+	err = vdt.Set(val)
+	if err != nil {
+		return fmt.Errorf("setting value to varying data type: %w", err)
+	}
+
+	*ad = ApprovalDistribution(vdt)
+	return nil
+}
```

Comment:
Can you check if just adding `Index` method here is sufficient? These creates a lot of repetitive methods.

Reply:
Removed unneeded functions.